fails to compile on mac

I tried master branch on Mac (saxpy.cu repro form https://github.com/jaredhoberock/agency/issues/134) , and get compilation error:

    $ nvcc saxpy.cu -I.  -std=c++11 -run
    ./agency/cuda/future.hpp:256:40: error: no member named 'when_all_events_are_ready' in namespace 'agency::cuda::detail'
    detail::event when_all_ready = detail::when_all_events_are_ready((futures.event())...);
                                   ~~~~~~~~^
    ./agency/cuda/detail/when_all_execute_and_select.hpp:285:32: error: no member named 'when_all_events_are_ready' in namespace
          'agency::cuda::detail'
    event when_all_ready = detail::when_all_events_are_ready(gpu, (outer_arg_future.event()), (agency::detail::get< TupleIndices> (tuple_of_fu...
                           ~~~~~~~~^
    2 errors generated.

    $ nvcc --version
    nvcc: NVIDIA (R) Cuda compiler driver
    Copyright (c) 2005-2015 NVIDIA Corporation
    Built on Sat_Jan__2_22:14:02_CST_2016
    Cuda compilation tools, release 8.0, V8.0.7

    $ git bisect
    ...
    3370470e39cb7575093fb1fe27b77212f0b0d087 is the first bad commit
    commit 3370470e39cb7575093fb1fe27b77212f0b0d087
    Author: Jared Hoberock <jaredhoberock@gmail.com>
    Date:   Tue Nov 3 15:20:01 2015 -0800
    
        Implement when_all() for cuda::detail::event

    :040000 040000 18274f62bd47f93f36e2897b4caa7038b0753505 573afd0a3f43155ee2119cbc313416f41eb2b989 M      agency

--- a/agency/cuda/detail/event.hpp
+++ b/agency/cuda/detail/event.hpp
@@ -231,26 +231,36 @@ class event
 
     template<class... Events>
     __host__ __device__
-    friend event when_all_events_are_ready(const gpu_id& gpu, Events&... events)
-    {
-      detail::stream s{gpu};
-
-      // tell the stream to wait on all the events
-      swallow(events.stream_wait(s)...);
-
-      // return a new event recorded on the stream
-      return event(std::move(s));
-    }
+    friend event when_all_events_are_ready(const gpu_id& gpu, Events&... events);
 
     template<class... Events>
     __host__ __device__
-    friend event when_all_events_are_ready(Events&... events)
-    {
-      // just use the current gpu
-      // XXX we might prefer the gpu associated with the first event
-      return when_all_events_are_ready(current_gpu(), events...);
-    }
+    friend event when_all_events_are_ready(Events&... events);
 };
+
+
+template<class... Events>
+__host__ __device__
+event when_all_events_are_ready(const gpu_id& gpu, Events&... events)
+{
+  detail::stream s{gpu};
+
+  // tell the stream to wait on all the events
+  event::swallow(events.stream_wait(s)...);
+
+  // return a new event recorded on the stream
+  return event(std::move(s));
+}
+
+
+template<class... Events>
+__host__ __device__
+event when_all_events_are_ready(Events&... events)
+{
+  // just use the current gpu
+  // XXX we might prefer the gpu associated with the first event
+  return agency::cuda::detail::when_all_events_are_ready(current_gpu(), events...);
+}
 
 
 } // end detail

